src/parquet/arrow/schema.cc

```diff
@@ -238,40 +277,54 @@ Status NodeToList(const GroupNode* group, TypePtr* out) {
 }
 
 Status NodeToField(const NodePtr& node, std::shared_ptr<Field>* out) {
-  std::shared_ptr<::arrow::DataType> type;
+  return NodeToFieldInternal(node, nullptr, out);
+}
+
+Status NodeToFieldInternal(const NodePtr& node,
+    const std::unordered_set<NodePtr>* included_leaf_nodes, std::shared_ptr<Field>* out) {
+
+  std::shared_ptr<::arrow::DataType> type = nullptr;
   bool nullable = !node->is_required();
 
+  *out = nullptr;
+
   if (node->is_repeated()) {
     // 1-level LIST encoding fields are required
     std::shared_ptr<::arrow::DataType> inner_type;
-    const PrimitiveNode* primitive = static_cast<const PrimitiveNode*>(node.get());
-    RETURN_NOT_OK(FromPrimitive(primitive, &inner_type));
-    auto item_field = std::make_shared<Field>(node->name(), inner_type, false);
-    type = ::arrow::list(item_field);
-    nullable = false;
+    if (IsIncludedLeaf(static_cast<NodePtr>(node), included_leaf_nodes)) {
+      const PrimitiveNode* primitive = static_cast<const PrimitiveNode*>(node.get());
```

Comment:
Hi,  I think this doesn't handles the repeated group field. For example,
```
message TestProtobuf.ArrayWithNestedGroupAndArray {
  optional int32 primitive;
  repeated group myComplex {
    optional int32 id;
    repeated group repeatedMessage {
      optional int32 someId;
    }
  }
}
```
So, I made an attempt to fix that issue, but didn't report earlier. The fix is shown as below against old code.
``` cpp
  if (node->is_repeated()) {
    // 1-level LIST encoding fields are required
    std::shared_ptr<::arrow::DataType> inner_type;
    if (node->is_group()) {
      const GroupNode* group = static_cast<const GroupNode*>(node.get());
      RETURN_NOT_OK(StructFromGroup(group, &inner_type));
    } else {
      const PrimitiveNode* primitive = static_cast<const PrimitiveNode*>(node.get());
      RETURN_NOT_OK(FromPrimitive(primitive, &inner_type));
    }
    auto item_field = std::make_shared<Field>(node->name(), inner_type, false);
    type = ::arrow::list(item_field);
    nullable = false;
```
Hope this helps.